packages/hardhat-core/test/internal/hardhat-network/provider/modules/evm.ts

```diff
@@ -1437,5 +1437,41 @@ describe("Evm module", function () {
         });
       });
     });
+
+    describe(`${name} provider (allowBlocksWithSameTimestamp)`, function () {
+      setCWD();
+      useProvider({ allowBlocksWithSameTimestamp: true });
```

Comment:
This is a bit nitpicky, but i think we should have another test that mines two blocks in a row, without manually altering the time, and they should get the same timestamp.

The reason I think this is important, is that this is the most common behavioral change that this pr introduces.

Reply:
Please ignore everything I said. This is already present in a different file.